examples/gc.ml

```diff
@@ -0,0 +1,99 @@
+open Lwt.Syntax
+
+(** make schema *)
+
+module Schema = Irmin.Schema.KV (Irmin.Contents.String)
+
+(** define configuration parameters for pack store *)
+
+module Store_conf : Irmin_pack.Conf.S = struct
+  (** number of entries in the internal inode B-tree. 32 performs well *)
+  let entries = 32
+
+  (** set this to 0 to only use inode hashing (this paraemter exists for
+      particular backwards compatibility applications with Irmin 1.0 data) *)
+  let stable_hash = 0
+
+  (** GC only supports minimal indexing strategy so [contents_length_header]
+      must not be [None], which requires an always indexing strategy *)
+  let contents_length_header = Some `Varint
+
+  (** this is the recommended sorting algorithm for new stores *)
+  let inode_child_order = `Hash_bits
+
+  (** throw an error (or not) if an empty tree is persisted *)
+  let forbid_empty_dir_persistence = true
+end
+
+(** make pack store *)
+
+module Maker = Irmin_pack_unix.Maker (Store_conf)
+module Store = Maker.Make (Schema)
+
+(** define configuration for repo *)
+
+module Repo_config : sig
+  val config : Irmin.config
+end = struct
+  (** how to handle large writes *)
+  let merge_throttle = `Block_writes
+
+  (** must use minimal indexing strategy to use GC *)
+  let indexing_strategy = Irmin_pack.Indexing_strategy.minimal
+
+  (** setting to true trades off performance for durability *)
+  let use_fsync = false
+
+  (** buffer size that triggers auto flushing to disk *)
+  let dict_auto_flush_threshold = 1_000_000
+
+  (** buffer size that triggers auto flushing to disk *)
+  let suffix_auto_flush_threshold = 1_000_000
+
+  (** location on disk to save the repository *)
+  let root = "/tmp/irmin-pack-example"
```

Comment:
There are problems using /tmp with the ci infrastructure I believe. I think we tend to use a local directory "./tmp/irmin-pack-example" instead.

Reply:
Good to know. Thanks!